numpy/core/src/umath/ufunc_type_resolution.c

```diff
@@ -63,6 +84,120 @@ raise_binary_type_reso_error(PyUFuncObject *ufunc, PyArrayObject **operands) {
     return -1;
 }
 
+/** Helper function to raise UFuncNoLoopError */
```

Comment:
If we go with the always return `-1` route, could you add a comment stating that it does so independent of whether it succeeds to raise the requested error or not? (I still prefer `0` on successfully raising, but not a big deal.)

Reply:
I'm using `-1` to mean "raised any error", not "raised the error I was expecting". Distinguishing those two cases isn't useful here anyway - if you errored out while trying to raise an error, it's probably not going to go well when you try to raise another one to say that the first one went wrong.

I can see the argument for making these `void`, but it seemed easier to match the existing functions.

Reply:
OK, that logic works for me too.

Reply:
I guess I resolved a bit prematurely - I think it would be helpful to note that the function returns `-1` unconditionally (here and below; otherwise everything looks good).

Reply:
Added that comment